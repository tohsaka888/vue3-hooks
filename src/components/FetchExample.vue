<script setup lang="ts">
import useFetch from '../hooks/useFetch';
import { baseUrl } from '../../config/index'
import { toRef } from 'vue';
const banners = useFetch<any>(`${baseUrl}/banner?type=1`, {
  method: 'GET'
})
console.log(banners)

console.log(banners?.value)

</script>

<template>
  <div v-if="banners?.data" v-for="item in banners?.data.banners">
    <img :src="item.pic" alt="item.pic" />
  </div>
</template>

<style scoped>
a {
  color: #42b983;
}

label {
  margin: 0 0.5em;
  font-weight: bold;
}

code {
  background-color: #eee;
  padding: 2px 4px;
  border-radius: 4px;
  color: #304455;
}
</style>
