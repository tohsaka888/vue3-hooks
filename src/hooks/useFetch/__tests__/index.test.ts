import { mount } from "@vue/test-utils";
import FetchExample from "../../../components/FetchExample.vue";

describe("Component", () => {
  test("something", () => {
    const wrapper = mount(FetchExample);
    expect(wrapper.element).toMatchSnapshot();
  });
});
