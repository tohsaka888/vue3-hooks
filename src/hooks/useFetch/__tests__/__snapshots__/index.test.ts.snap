// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component something 1`] = `
<p>
   Recommended IDE setup: 
  <a
    href="https://code.visualstudio.com/"
    target="_blank"
  >
    VSCode
  </a>
   + 
  <a
    href="https://github.com/johnsoncodehk/volar"
    target="_blank"
  >
    Volar
  </a>
</p>
`;
